src/app/services/wallet.service.ts

```diff
@@ -177,22 +178,22 @@ export class WalletService {
   unlockWallet(wallet: Wallet, seed: string): Promise<void> {
     seed = this.getCleanSeed(seed);
 
-    return new Promise<void>((resolve) => {
-      let currentSeed = this.convertAsciiToHexa(seed);
-      wallet.addresses.forEach(address => {
-        const fullAddress = this.cipherProvider.generateAddress(currentSeed);
-        if (fullAddress.address !== address.address) {
-          throw new Error(this.translate.instant('service.wallet.wrong-seed'));
-        }
-        address.next_seed = fullAddress.next_seed;
-        address.secret_key = fullAddress.secret_key;
-        address.public_key = fullAddress.public_key;
-        currentSeed = fullAddress.next_seed;
-      });
+    return new Promise<void>((resolve, reject) => {
+      const currentSeed = this.convertAsciiToHexa(seed);
 
-      wallet.seed = seed;
-      this.updateWallet(wallet);
-      return resolve();
+      WebWorkersHelper.ExcecuteWorker('/assets/scripts/workers/generate-addresses.worker.js', { seed: currentSeed, addresses: wallet.addresses })
```

Comment:
I think that `WebWorkersHelper.ExcecuteWorker` should be in `CipherProvider.generateAddress` and not here. I think so because with this change we now have two different codes to generate addresses: the one in `CipherProvider.generateAddress` and the one in `generate-addresses.worker.js`. To not have duplicate code, it is necessary to choose one of the two, but `generate-addresses.worker.js` has some problems, like the fact that it has no tests (while `CipherProvider` has unit tests and the test suite adapted from Go) and it is in Javascript (which is inconvenient due to the lack of consistency with the rest of the code, the lack of compilation errors, etc.).

Bearing that in mind, and the fact that it will eventually be necessary to also adapt `CipherProvider.prepareTransaction`, I think that keping the code in the `.worker.js` files to the bare minimun (only to call the functions from `main.js`) and keep all the logic inside `CipherProvider` is much more convenient.

Reply:
Would it be OK to have 2 separate web workers: 1- generate single address; 2 - generate the list of addresses?

Reply:
I'm not sure why that would be needed. `CipherProvider` does not have a function to create a list of addresses, it only creates one address at a time. `Cipher` from `main.js` have a function called `GenerateAddresses` but it just creates one address.

If you want to create a new function that returns multiple addresses, you can simply use TypeScript to call several times the web worker that creates one address, that would not negatively affect the UX, so it would not be necessary to create a web worker specifically for that. The idea is to write as little JavaScript code as possible.